Source/WebKit/WebProcess/WebPage/WebPage.messages.in

```diff
@@ -237,6 +237,7 @@ GenerateSyntheticEditingCommand(enum:uint8_t WebKit::SyntheticEditingCommandType
     GetSelectionAsWebArchiveData() -> (std::optional<IPC::SharedBufferReference> data)
     GetSourceForFrame(WebCore::FrameIdentifier frameID) -> (String string)
     GetWebArchiveOfFrame(WebCore::FrameIdentifier frameID) -> (std::optional<IPC::SharedBufferReference> dataReference)
+    GetWebArchiveOfFrameWithFileName(WebCore::FrameIdentifier frameID, String fileName) -> (std::optional<IPC::SharedBufferReference> dataReference)
```

Comment:
So this new message will send a whole webarchive over CoreIPC.

@beidson and/or @achristensen07 should definitely weigh in on this part.

I wonder if we can't do all the writing in the WebContent process (not the UI process--if I understand the PR correctly).

Reply:
Sihui reminded me that we're already sending webarchive plists over CoreIPC for...saving WebArchives, so this really isn't new (just a new endpoint).